tiledb/sm/misc/test/unit_mgc_dict.cc

```diff
@@ -471,35 +453,10 @@ int embedded_vs_external_identifications() {
     return 0;
   };
 
-  if (auto rval = proc_list(file_data_sizes1, true); rval != 0)
-    return rval;
-  if (auto rval = proc_list(file_data_sizes2, true); rval != 0)
-    return rval;
-
-  if (auto rval = proc_list(file_data_sizes1, false); rval != 0)
-    return rval;
-  if (auto rval = proc_list(file_data_sizes2, false); rval != 0)
-    return rval;
-
-  if (errcnt) {
-    fprintf(
-        stderr, "%d mismatch errors magic embedded vs external data\n", errcnt);
-  }
-
-  return errcnt ? 99 : 0;
-}
-
-int main() {
-  int failures = 0;
+  REQUIRE(proc_list(file_data_sizes1, true) == 0);
+  REQUIRE(proc_list(file_data_sizes2, true) == 0);
+  REQUIRE(proc_list(file_data_sizes1, false) == 0);
+  REQUIRE(proc_list(file_data_sizes2, false) == 0);
 
-  failures |= check_embedded_data_validity();
-  failures |= embedded_vs_external_identifications();
-
-  if (failures) {
-    fprintf(stderr, "\nERRORS mgc_dict validation\n");
-    return 1;
-  } else {
-    fprintf(stderr, "NO errors encountered in mgc_dict validation\n");
-    return 0;
-  }
+  REQUIRE(errcnt == 0);
```

Comment:
We won't need this if all the tests to which it might apply are executed with `CHECK` or `REQUIRE`. That will take some more rewrites.

Reply:
Updated. We now make a proper `CHECK` instead of keeping the count of errors.